cocotb/triggers.py

```diff
@@ -456,13 +456,21 @@ def __repr__(self):
 class Lock(object):
     """Lock primitive (not re-entrant).
 
-    This should be used as::
+    This can be used as::
 
-        yield lock.acquire()
+        await lock.acquire()
         try:
             # do some stuff
         finally:
             lock.release()
+
+    The lock can be used as an asynchronous
+    context manager in a :keyword:`with`,
+    statement, provided Python 3.5 or newer is
+    used::
```

Comment:
is :keyword:`async with` supported? Also we can drop the version statement. Python 3.5 is already the lowest supported version.

Reply:
Seemingly yes. Good idea